### PR TITLE
fix: support the latest docker-compose 1.27 spec

### LINKED ISSUE #9269: Latest docker-compose 1.27 specification is not supported
**What Renovate type, platform and version are you using?**

GitHub App

**Describe the bug**

Since docker-compose version 1.27, the `version:` top-level Element is no longer required for a valid docker-compose.yaml file (https://github.com/compose-spec/compose-spec/blob/master/spec.md#version-top-level-element).

See also https://docs.docker.com/compose/: `version: "3.9"  # optional since v1.27.0`.

However, the presence of the version element is currently used to figure out whether version 1 or 2 of the specification is used, as the services are then either defined at a top-level or within the `services` key: https://github.com/renovatebot/renovate/blob/master/lib/manager/docker-compose/extract.ts#L73-L76

With docker-compose 1.27 we now have the case where the `version` element may be missing, but the services are still defined within a `services` section, preventing renovate from finding the `image`s in the correct location.

**Relevant debug logs**

**Have you created a minimal reproduction repository?**

https://github.com/chgl/renovate-docker-compose-v127/pull/2/files

- [ ] This is a really small bug, it does not need a reproduction (think small typo)
- [x] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

I think replacing the logic in https://github.com/renovatebot/renovate/blob/master/lib/manager/docker-compose/extract.ts#L73-L76 for a simple check for the presence of a `services` key should work.

### PR DESCRIPTION
**Pull request by chgl**
## Changes:

This changes the way the list of services in a docker-compose.yaml file are retrieved. Previously the presence of the `version:` key was used to determine whether the services are part of the top-level file or are within a `services:` key. Since `version` is optional since docker-compose spec v1.27, this now simply uses `document.services` if set, or `document` if not.

## Context:

Closes #9269

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository (see https://github.com/chgl/renovate-docker-compose-v127/pull/2/files)

### CODE REVIEW

File: lib/manager/docker-compose/extract.spec.ts [hunk 1]
```diff
@@ -11,6 +11,11 @@ const yamlFile3 = readFileSync(
   'utf8'
 );
 
+const yamlFile3NoVersion = readFileSync(
+  'lib/manager/docker-compose/__fixtures__/docker-compose.3-no-version.yml',
+  'utf8'
+);
+
 describe('lib/manager/docker-compose/extract', () => {
```
Comment (viceice) [inline lib/manager/docker-compose/extract.spec.ts:19]:
```suggestion
describe(getName(__filename), () => {
```

import from `test/util` 😏

**Review by viceice — COMMENTED** [review]
just a small thing

File: lib/manager/docker-compose/extract.spec.ts (hunk 1 above)
Comment (chgl) [reply lib/manager/docker-compose/extract.spec.ts:19]: oh, fancy! Thank you!

**Review by viceice — COMMENTED** [review]
needs snapshot update

**Review by viceice — APPROVED** [review]

**Pull request merged**